src/test/scala/nl/biopet/tools/tenxkit/mergebams/MergeBamsTest.scala

```diff
@@ -32,4 +37,204 @@ class MergeBamsTest extends ToolTest[Args] {
+                       "bam2-wgs2-lib1")
+    it2.close()
+    reader.close()
+  }
+
+  @Test
+  def testDuplets(): Unit = {
+    val outputBam = File.createTempFile("merge.", ".bam")
+    outputBam.deleteOnExit()
+    val outputBarcodes = File.createTempFile("merge.", ".txt")
+    outputBarcodes.deleteOnExit()
+    MergeBams.main(
+      Array(
+        "--inputBam",
+        s"bam1=${resourcePath("/wgs2.realign.bam")}",
+        "--inputBarcode",
+        s"bam1=${resourcePath("/wgs2.readgroups.txt")}",
+        "--inputBam",
+        s"bam2=${resourcePath("/wgs2.realign.bam")}",
+        "--inputBarcode",
+        s"bam2=${resourcePath("/wgs2.readgroups.txt")}",
+        "--outputBam",
+        s"$outputBam",
+        "--outputBarcodes",
+        s"$outputBarcodes",
+        "--duplets",
+        "1",
+        "--sampleTag",
+        "RG",
+        "--umiTag",
+        "MD"
+      ))
+
+    val reader = SamReaderFactory
+      .makeDefault()
+      .validationStringency(ValidationStringency.SILENT)
+      .open(outputBam)
+    val it1 = reader.iterator()
+    it1.size shouldBe 60000
+    it1.close()
+
+    val it2 = reader.iterator()
+    val cells =
+      it2.flatMap(x => Option(x.getAttribute("RG")).map(_.toString)).toSet
+    cells shouldBe Set("bam1-wgs2-lib2",
+                       "bam1-wgs2-lib1",
+                       "bam2-wgs2-lib2",
+                       "bam2-wgs2-lib1",
+                       "bam1-wgs2-lib1_bam2-wgs2-lib1")
+    it2.close()
+    reader.close()
+  }
+
+  @DataProvider(name = "fractions")
+  def fractions: Array[Array[Any]] =
+    Array(Array(0.1),
+          Array(0.2),
+          Array(0.3),
+          Array(0.4),
+          Array(0.5),
+          Array(0.6),
+          Array(0.7),
+          Array(0.8),
+          Array(0.9),
+          Array(1.0))
+
+  @Test(dataProvider = "fractions")
+  def testDownsample(fraction: Double): Unit = {
+    val outputBam = File.createTempFile("merge.", ".bam")
+    outputBam.deleteOnExit()
+    val outputBarcodes = File.createTempFile("merge.", ".txt")
+    outputBarcodes.deleteOnExit()
+    MergeBams.main(
+      Array(
+        "--inputBam",
+        s"bam1=${resourcePath("/wgs2.realign.bam")}",
+        "--inputBarcode",
+        s"bam1=${resourcePath("/wgs2.readgroups.txt")}",
+        "--inputBam",
+        s"bam2=${resourcePath("/wgs2.realign.bam")}",
+        "--inputBarcode",
+        s"bam2=${resourcePath("/wgs2.readgroups.txt")}",
+        "--outputBam",
+        s"$outputBam",
+        "--outputBarcodes",
+        s"$outputBarcodes",
+        "--sampleTag",
+        "RG",
+        "--downsampleFraction",
+        fraction.toString
+      ))
+
+    val reader = SamReaderFactory
+      .makeDefault()
+      .validationStringency(ValidationStringency.SILENT)
+      .open(outputBam)
+    val it1 = reader.iterator()
+    val aim = 40000 * fraction
+    val reads = it1.size
+    // the 0.02 means that the result of the downsampling is variable with 2%
```

Comment:
You may be either too lenient or to strict here with the 2% here depending on the fraction chosen. 
This means tests will randomly fail during CI.

It is best to set up the boundaries using a binomial distribution and test whether the aim is within those boundaries.